Src/Base/AMReX_MFIter.cpp

```diff
@@ -267,15 +271,6 @@ MFIter::Initialize ()
             "Nested or multiple active MFIters is not supported by default.  This can be changed by calling MFIter::allowMultipleMFIters(true)".);
     }
 
-#ifdef AMREX_USE_GPU
-    if (device_sync) {
-#ifdef AMREX_USE_OMP
-#pragma omp single
-#endif
-        Gpu::streamSynchronize();
```

Comment:
I think this is still necessary for the case that MFIter is going to use a different stream than the stream used outside of MFIter.
Example:

1. expensive async memcpy on stream 0
2. important data async memcpy on stream 0
3. MFIter loop starts with two streams
3.1. kernal using the important data on stream 0
3.2. another kernal using the important data on stream 1
4. MFIter loop ends

The issue is that 3.2 can start before 1 is finished because there was no synchronize, however it will produce wrong results because 2 is not finished jet. Notice that if 3.2 where to use stream 0 then no synchronize in both the constructor or destructor of MFIter would be required. This is what is done in HiPACE++ which is greatly beneficial for low resolutions.

Reply:
In principle, yes, that could happen.  In practice, I have never seen that much overlap between streams that could cause issues.

We used to use the special null stream outside the kernel and we did not have the sync in the ctor of MFIter.  However, SYCL/DPC++ does not have the equivalent of the null stream.  So the behavior was not consistent before we switched to avoiding the null stream.  It was because of the concern you raised that we added a sync in the ctor MFIter.  After thinking about this for a long time, I think it's better to not do the sync in the ctor.

Reply:
For MAC projection in an amr-wind test with max_grid_size=32, this PR improves the time from 14.26 to 11.99.  The overall time went down from 56.04 to 48.92.

Reply:
Anyway, if there is data movement before MFIter, it's now up to the user to make sure that does not cause an issue.  The user could insert an explicit sync before MFIter starts or at the end of iteration 0.

Reply:
@AlexanderSinn I tested the scenario you have described. It did happen when the memcopy in (1) was very expensive.  So now the user had better make sure the memcpy finishes before MFIter starts.

Reply:
Good to know

Reply:
Let me test if I put it back, how much performance penalty there is for the amr-wind test.  Maybe all the other changes made in this PR is all that matters.

Reply:
If (1) and (2) are swapped, then I don't see any issues.  This makes sense.

Reply:
It may also be worth trying to have only one stream and no synchronize in the dtor or ctor. This should work in most cases except when code relies on the synchronize in the ctor for e.g. dtoh memcpy.

Reply:
In many cases, use multiple streams is faster than a single stream.

I am thinking of adding something that can be used to disable syncs and specify the number of streams in a given scope.

Reply:
The good news is that I do not see much difference if the sync in MFIter ctor is on in two tests with a small max grid size on summit and crusher.  So the improvement in this PR mostly comes from other changes.  I will put the sync back for safety.

Reply:
@AlexanderSinn Thanks for your constructive feedback!